### PR TITLE
Add update count statistics in density test

### PR DESCRIPTION
**Pull request by Random-Liu**
In this PR, I added update count statistics in density test, so as to help us understand pod updates during pod batch creation.

Originally, this is part of #18410. I split this to a separate PR to make it easier for reviewing.

@yujuhong

### CODE REVIEW

**Comment by yujuhong** [comment]
LGTM!

**Comment by Random-Liu** [comment]
@yujuhong Fixed, PTAL. :)

**Comment by Random-Liu** [comment]
@k8s-bot e2e test this.

File: test/e2e/density.go [hunk 1]
```diff
@@ -250,6 +250,36 @@ var _ = Describe("Density [Skipped]", func() {
...
+					AddFunc: func(_ interface{}) {
+						updateCount++
+					},
+					UpdateFunc: func(_, _ interface{}) {
+						updateCount++
+					},
+					DeleteFunc: func(_ interface{}) {
+						updateCount++
+					},
+				},
```
Comment (yujuhong) [inline test/e2e/density.go]: Why couting both Add and Delete?

Comment (Random-Liu) [inline test/e2e/density.go]: I should remove them to just count "Update" count, it should be more meaningful for status manager. :)

Comment (Random-Liu) [inline test/e2e/density.go]: Done

File: test/e2e/density.go [hunk 2]
```diff
@@ -250,6 +250,30 @@ var _ = Describe("Density [Skipped]", func() {
 			stop := make(chan struct{})
 			go controller.Run(stop)
 
+			// Create a listener for api updates
+			updateCount := 0
```
Comment (yujuhong) [inline test/e2e/density.go]: As we discussed offline, `updateCount` is accessed by multiple goroutines and will need to be fixed.

Comment (Random-Liu) [inline test/e2e/density.go]: ACK. :)

Comment (Random-Liu) [inline test/e2e/density.go]: DONE

**Comment by Random-Liu** [comment]
@k8s-bot test this please.

**Pull request merged**